### PR TITLE
Set `accelerator="auto"` by default in lightning estimators

### PR DESCRIPTION
**Pull request by lostella**
*Description of changes:* Set `accelerator="auto"` by default in lightning estimators.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
We could backport this I guess?

**Comment by abdulfatir** [comment]
While we're at this, do you think these are also restrictive: https://github.com/awslabs/gluonts/blob/1cc17c1dbcbdbb9b9b931913541a21c25aebd740/src/gluonts/torch/model/simple_feedforward/estimator.py#L250-L252

**Review by abdulfatir — DISMISSED** [review]

File: src/gluonts/torch/model/estimator.py [hunk 1]
```diff
@@ -197,9 +197,12 @@ def train_model(
             monitor=monitor, mode="min", verbose=True
         )
 
-        custom_callbacks = self.trainer_kwargs.get("callbacks", [])
-        callbacks = [checkpoint] + custom_callbacks
-        trainer_kwargs = {**self.trainer_kwargs, "callbacks": callbacks}
+        custom_callbacks = self.trainer_kwargs.pop("callbacks", [])
+        trainer_kwargs = {
```
Comment (jaheba) [inline src/gluonts/torch/model/estimator.py]: Can we just call the trainer directly with this?

Comment (lostella) [reply src/gluonts/torch/model/estimator.py]: how do you mean?

File: src/gluonts/torch/model/estimator.py [hunk 2]
```diff
@@ -197,9 +197,12 @@ def train_model(
...
+            "accelerator": "auto",
+            "callbacks": [checkpoint] + custom_callbacks,
+            **self.trainer_kwargs,
+        }
         trainer = pl.Trainer(**trainer_kwargs)
```
Comment (jaheba) [inline src/gluonts/torch/model/estimator.py]:
```suggestion
        trainer = pl.Trainer(**{
            "accelerator": "auto",
            "callbacks": [checkpoint] + custom_callbacks,
            **self.trainer_kwargs,
        })
```

**Review by jaheba — DISMISSED** [review]

**Review by jaheba — APPROVED** [review]
